src/helpers/OnSiteHoldRequestHelper.js

```diff
@@ -0,0 +1,161 @@
+    return OnSiteHoldRequestHelper.postOnSiteHoldRequests(
+      records,
+      apiData
+    )
+  },
+  generateModel: ({
+    patron,
+    nyplSource,
+    record,
+    pickupLocation,
+    neededBy,
+    numberOfCopies,
+    docDeliveryData,
+    id,
+    jobId,
+    requestType,
+  }) => {
+    return {
+      patron,
+      nyplSource,
+      record,
+      pickupLocation,
+      neededBy,
+      numberOfCopies,
+      docDeliveryData
+    };
+  },
+  postOnSiteHoldRequests: (requests, apiData) => {
+    const {
+      apiHelper,
+      apiKey,
+      apiBaseUrl,
+    } = apiData;
+    return new Promise((resolve, reject) => {
+      const onSiteHoldRequestResponses = async.mapSeries(requests, (request, callback) => {
+        const isEdd = request.docDeliveryData && request.docDeliveryData.emailAddress;
+        if (!isEdd) {
+          logger.info(`Not implemented: on-site hold request for on-site use`, { holdRequestId: request.id });
+          return ResultStreamHelper.postRecordToStream({
+            holdRequestId: request.id,
+            jobId: request.jobId,
+            errorType: 'not-implemented-use-on-site-request',
+            errorMessage: `Not implemented: on-site hold request for on-site use record ${request.id}`
+          })
+          .then(response => {
+            logger.info(
+              `successfully posted failed hold request record (${request.id}) to HoldRequestResult stream`,
+              { holdRequestId: request.id }
+            );
+
+            return callback(null, request);
+          })
+          .catch(error => {
+            logger.error(
+              `unable to post hold request for on-site record to results stream, received an error from HoldRequestResult stream; exiting promise chain due to fatal error`,
+              { holdRequestId: request.id, error: error }
+            );
+
+            // At this point, we could not POST the failed hold request to the results stream.
+            // We are exiting the promise chain and restarting the kinesis handler
+            return callback(HoldRequestConsumerError({
+              message: `unable to post failed hold request for on-site record to results stream, received error from HoldRequestResult stream; exiting promise chain due to fatal error`,
+              type: 'hold-request-result-stream-error',
+              status: error.response && error.response.status ? error.response.status : null,
+              function: 'postRecordToStream',
+              error: error
+            }));
+          });
+        }
+        const eddLogText = isEdd ? 'EDD ' : '';
+        const requestBody = OnSiteHoldRequestHelper.generateModel(request)
+        logger.info(`posting ${eddLogText}hold request for on-site item (${requestBody.record})`, { holdRequestId: request.id });
+        return axios.post(
+          `${apiBaseUrl}on-site-hold-requests`,
+          requestBody,
+          apiHelper.constructApiHeaders(apiKey)
```

Comment:
Seems a shame to pass the apiHelper instance from component to component just to gain access to `constructApiHelpers`. Shame that `apiHelper` does not provide more readily usable, generic functions for performing NYPL Platform API queries. I think - given the almost functional style of the app - your approach is pragmatic.